### DISCUSSION TITLE
Add code coverage reporting

### DISCUSSION THREAD

**Discussion by eNeRGy164** [fluentassertions/fluentassertions#1810] [Ideas]
I'm not sure what your stance is on adding code coverage to FluentAssertions, but I find it a good way to detect untested functionality.

Because I added code coverage with `coveralls` to one of my own projects, I created a quick demo how this could look for FA.
Example PR: https://github.com/eNeRGy164/fluentassertions/pull/2
Coveralls: https://coveralls.io/github/eNeRGy164/fluentassertions

I didn't point the PR to this repo the keep the CI reporting clean before knowing if you would want to go into this direction.

Because there is no baseline in coveralls, the information is a bit limited, and doesn't post in the PR thread yet.
It would look something like https://github.com/eNeRGy164/LivingDocumentation/pull/47
`Coveralls`: https://coveralls.io/github/eNeRGy164/LivingDocumentation

There are 2 ways of reporting in the PR, `coveralls` can do it on it's own (but only works after having a baseline), and by using `CodeCoverageSummary` & `sticky-pull-request-comment`.
The last one is in a separate commits as explanation. As it needs to run as a docker container, which in GitHub actions is only allowed for Linux, there is also a second stage added for this to the CI.

**Comment by jnyrup** [comment]
I'd love having code coverage enabled for this repo.
Mostly to ensure that new functionality is properly tested.
I believe it could ease reviewing PRs in a similar way that static analyzers do some of the work.

I'm not sure I understand the difference between the two ways of reporting?
Do you have any favorite
I've recently tested that the CI pipeline can run under Linux, so it should not block picking the latter if it's better.

I also had a thought about adding Linux to the CI pipeline anyways.

**Reply by eNeRGy164** [reply]
I've kept it to only using coveralls for the reporting, it keeps the build config much more simple for now, only having coveralls upload in it.
I think coveralls is useful for having a trend over time.

I moved report generator (which was needed for combining output anyway) to Nuke, so there will be a local report, and an `lcov` if you are running as GitHub action

**Reply by dennisdoomen** [reply]
We've been using coverlet in most of our projects and all our TeamCity and GHA build agents run on Linux.
